packages/liveblocks-core/src/room.ts

```diff
@@ -2848,48 +2848,29 @@ export function createRoom<
 
     const PAGE_SIZE = 50;
 
-    const response = await httpClient2.fetch(
-      url`/v2/c/rooms/${config.roomId}/threads`,
-      undefined,
-      {
-        cursor: options?.cursor,
-        query,
-        limit: PAGE_SIZE,
-      }
-    );
-
-    if (response.ok) {
-      const json = await (response.json() as Promise<{
-        data: ThreadDataPlain<M>[];
-        inboxNotifications: InboxNotificationDataPlain[];
-        deletedThreads: ThreadDeleteInfoPlain[];
-        deletedInboxNotifications: InboxNotificationDeleteInfoPlain[];
-        meta: {
-          requestedAt: string;
-          nextCursor: string | null;
-        };
-      }>);
-
-      return {
-        threads: json.data.map(convertToThreadData),
-        inboxNotifications: json.inboxNotifications.map(
-          convertToInboxNotificationData
-        ),
-        nextCursor: json.meta.nextCursor,
-        requestedAt: new Date(json.meta.requestedAt),
-      };
-    } else if (response.status === 404) {
-      return {
-        threads: [],
-        inboxNotifications: [],
-        deletedThreads: [],
-        deletedInboxNotifications: [],
-        nextCursor: null,
-        requestedAt: new Date(),
```

Comment:
This code we had here looks a little bit weird to me so maybe indeed it's fine to handle it, but I just want to confirm there isn't an edge case we're overlooking with this and changing in an unexpected way. If this 404s, it most likely means the room does not exist, which would happen either if the room actually does not exist, or it was deleted.

In the new code, this edge case would throw an HttpError. Maybe this is fine, but a bit hard to tell without understanding the reason this code existed. Either way, if the code is still relevant, I would really love to see a unit test for this, just to capture this required behavior deliberately, which is currently not the case.